File: 0704_search.cpp
#include<iostream>
#include<vector>
using namespace std;

int serch(vector<int> nums, int target) {
    int l = 0; 
    int r = nums.size();
    while (l < r) {
        int mid = l + ((r - l) / 2);
        if (nums[mid] > target) {
            r = mid;
        }
        else if (nums[mid] < target) {
            l = mid + 1;
        }
        else {
            return mid;
        }
    }
    return -1;
}
#ACM输入
int main() {
    int n;
    vector<int> nums;
    int target;
    cin >> n;
    nums.resize(n);
    for (int i = 0;i < n; i++) {
        cin >> nums[i];
    }
    cin >> target;
    int result = serch(nums, target);
    cout << result;

}

File: problem/0704_search.py
def search(nums, target):
    l, r = 0, len(nums) - 1
    while l <= r:
        mid = l + (r - l) // 2
        if nums[mid] < target:
            l = mid + 1
        elif nums[mid] > target:
            r = mid - 1
        else:
            return mid
    return -1


#n = int(input())
nums = list(map(int, input().split()))

target = int(input())
result = search(nums, target)
print(result)
